WordPress/Classes/Models/BasePost.m

```diff
@@ -219,4 +219,51 @@ - (NSString *)statusForDisplay
     return statusText;
 }
 
+#pragma mark - Status
+
+- (BOOL)isDraft
+{
+    return [self.status isEqualToString:BasePostStatusDraft];
+}
+
+- (BOOL)isPending
+{
+    return [self.status isEqualToString:BasePostStatusPending];
+}
+
+- (BOOL)isPublished
+{
+    return ([self.status isEqualToString:BasePostStatusPublish]
+            && [self hasRemote]);
```

Comment:
Not sure why having a remote would be a condition of having a publish post status.